crates/languages/src/go.rs

```diff
@@ -505,33 +505,36 @@ impl ContextProvider for GoContextProvider {
         _: Option<Arc<dyn language::File>>,
         _: &AppContext,
     ) -> Option<TaskTemplates> {
+        let package_cwd = if GO_PACKAGE_TASK_VARIABLE.template_value() == "." {
+            None
+        } else {
+            Some("$ZED_DIRNAME".to_string())
+        };
+
         Some(TaskTemplates(vec![
             TaskTemplate {
                 label: format!(
                     "go test {} -run {}",
                     GO_PACKAGE_TASK_VARIABLE.template_value(),
                     VariableName::Symbol.template_value(),
                 ),
-                command: "go".into(),
-                args: vec![
-                    "test".into(),
-                    GO_PACKAGE_TASK_VARIABLE.template_value(),
-                    "-run".into(),
-                    format!("^{}\\$", VariableName::Symbol.template_value(),),
```

Comment:
@mrnugget Why remove the regular expression here? For go test, -run=xxx will run all unit tests whose names match xxx. That means if I have two unit tests, TestXXX and TestXXX1, -run=TestXXX will run both of them instead of just running TestXXX.

Reply:
@glaudiston @mrnugget 
![image](https://github.com/user-attachments/assets/82e67d3c-0b11-4f7c-b753-355becb054a7)

I ran TestXXX, I expect success but not.

Reply:
Oh! That was a mistake. I think I thought I simplified the label, but I modified the command, not the label. Will fix!